shiny-api/updater/updater.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"log"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/apiarian/migration-playground/shiny-api/common"
+)
+
+type Updater struct {
+	kc            *common.KafkaClient
+	command_topic string
+	done          chan struct{}
+	assignIDs     bool
+	nextID        int
+	mux           *sync.Mutex
+	entityCache   map[string]*common.Thing
+}
+
+func NewUpdater(
+	kc *common.KafkaClient,
+	command_topic string,
+	assignIDs bool,
+	nextID int,
+) *Updater {
+	return &Updater{
+		kc:            kc,
+		command_topic: command_topic,
+		done:          make(chan struct{}),
+		assignIDs:     assignIDs,
+		nextID:        nextID,
+		mux:           &sync.Mutex{},
+		entityCache:   make(map[string]*common.Thing),
```

Comment:
We've found it unlikely in the entities we deal with, that you should keep them in memory.  For this example its probably ok (and maybe if your service only deals with things down to the agencies level you'll never deal with a dataset that is big).